### PR TITLE
Add macos to the CI

### PR DESCRIPTION
**Pull request by euri10**
Slighlty different than https://github.com/encode/uvicorn/pull/1395 @Kludex , let me know what you think.
In particular it shrink the socket name in macos CI and solves the issues you had on the `test_fragmentation` for macos

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -136,3 +141,38 @@ def reload_directory_structure(tmp_path_factory: pytest.TempPathFactory):
 @pytest.fixture(scope="function")
 def logging_config() -> dict:
     return deepcopy(LOGGING_CONFIG)
+
+
+@pytest.fixture
+def short_socket_name(tmp_path, tmp_path_factory):  # pragma: py-win32
+    max_sock_len = 100
```
Comment (Kludex) [inline tests/conftest.py:148]: I'm usually against comments, but a comment explaining the reason for the number 100 would be cool.

**Review by Kludex — APPROVED** [review]
Just a comment about the number 100, but other than that, this is great.

Thanks! :)

**Comment by euri10** [comment]
> Just a comment about the number 100, but other than that, this is great.
> 
> Thanks! :)

I've taken it from here https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars/367012#367012

we could have taken 104, but some new new bsd implementations go as low as 92 bytes.
100 seemed a good compromise :)

**Comment by Kludex** [comment]
Yep, I'm aware, but I meant a comment because it will be easier to grasp in code what the number represents without having to open the PR.

**Comment by euri10** [comment]
oups I didnt understand sorry :)

**Pull request merged**